### ISSUE TITLE
Cake.dll does not run on .Net Core 2.0

### ISSUE THREAD

**Issue by secana** [cake-build/cake#1781]
### What You Are Seeing?
We are getting the error:

```
It was not possible to find any compatible framework version
The specified framework 'Microsoft.NETCore.App', version '1.0.4' was not found.
  - Check application dependencies and target a framework version installed at:
      /
  - Alternatively, install the framework version '1.0.4'.
```

when we run `dotnet Cake.dll` under Ubuntu 16.04 with only .Net Core 2.0 installed.

### What is Expected?
That `dotnet Cake.dll` runs with only .Net Core 2.0 installed

### What version of Cake are you using?
0.21.1

### Are you running on a 32 or 64 bit system?
x64

### What environment are you running on?  Windows? Linux? Mac?
Linux (Ubuntu 16.04 server LTS)

### Are you running on a CI Server?  If so, which one?
No

### How Did You Get This To Happen? (Steps to Reproduce)
Download the Cake.CoreCLR.0.21.1.zip and unpack it.
Than run the `dotnet Cake.dll` command in the unpacked folder.

### Output Log

```
user@foo: dotnet Cake.dll
It was not possible to find any compatible framework version
The specified framework 'Microsoft.NETCore.App', version '1.0.4' was not found.
  - Check application dependencies and target a framework version installed at:
      /
  - Alternatively, install the framework version '1.0.4'.
```

**Comment by adamhathcock** [comment]
I'm running into this too.

This is because Cake.dll targets only `netcoreapp1.0` and `net461`

A `netcoreapp2.0` target should be added as it's an executable,

**Comment by adamhathcock** [comment]
If I attempt to add a `netcoreapp2.0` target to Cake.dll, would that be accepted? @devlead @patriksvensson @gep13 ?

**Comment by secana** [comment]
Very dirty workaround is to install netcore 1.0 and than install netcore 2.0 on the same machine. So the Cake script runs with netcore 1.0 and builds for netcore 2.0 but that shouldn't be the solution to the problem.

**Comment by adamhathcock** [comment]
I don't want to install old tech :)

I've started to play with targeting `netcoreapp2.0` and `netstandard2.0` (mainly because some dependences are different based on moving to 2.0) in a branch.  Since `net461` is the only full framework supported on cake.exe, the other assemblies can be all `netstandard2.0` which is fun.  Right now, I'm targetting 1.6 and 2.0 on libs.  Tests do `net461` and `netcoreapp2.0` and cake.exe is `net461`, `netcoreapp1.0` and `netcoreapp2.0`

**Comment by ErikSchierboom** [comment]
@adamhathcock It would be great to have Cake be able to run on .NET Core 2.0!

**Comment by devlead** [comment]
> If I attempt to add a netcoreapp2.0 target to Cake.dll, would that be accepted?

@adamhathcock sure, netcoreapp2.0 is something we would like to support. Can't promise it will be accepted before I seen it though 😎 

Would also need to add integration tests with 2.0. 

Looking at what other tools are doing is to place framework monikers in nuget tools folders.  Core/Common shouldn't be needed till be touched as those are 1.6 and "should" work fine with 2.0.

**Comment by adamhathcock** [comment]
Of course with review :)

1.6 should be fine with 2.0.  There are awkwardnesses like assembly and appdomain things that don’t exist in 2.0 (because it looks like full framework now) but are needed for 1.6.  These require separate monikers.  I’m working through it slowly.

I’ll make sure I try to get all test places going.  So far no blockers.

**Comment by jackjwilliams** [comment]
Should I assume this project is dead since it's been going on 4 months and no update for .NET Core 2?

**Comment by joskoanicic** [comment]
This can be closed now with #1812 merged?  It would be nice to see an official v0.26.0 build.
